src/main/scala/nl.knaw.dans.easy.solr4files/ApplicationWiring.scala

```diff
@@ -15,15 +15,88 @@
  */
 package nl.knaw.dans.easy.solr4files
 
+import java.net.{ URI, URL }
+
+import nl.knaw.dans.easy.solr4files.components._
+import nl.knaw.dans.lib.error.{ CompositeException, TraversableTryExtensions }
 import nl.knaw.dans.lib.logging.DebugEnhancedLogging
 
+import scala.util.{ Failure, Success, Try }
+
 /**
  * Initializes and wires together the components of this application.
  *
  * @param configuration the application configuration
  */
-class ApplicationWiring(configuration: Configuration) extends DebugEnhancedLogging
-// Mix in components
-{
-  // Configure components with configuration settings.
+class ApplicationWiring(configuration: Configuration)
+  extends DebugEnhancedLogging with VaultIO with Vault with Solr {
+
+  // don't need resolve for solr, URL gives more early errors TODO perhaps not enough early errors
+  override val solrUrl: URL = new URL(configuration.properties.getString("solr.url", ""))
+  override val vaultBaseUri: URI = new URI(configuration.properties.getString("vault.url", ""))
+
+  def initAllStores(): Try[FeedBackMessage] = {
+    getStoreNames
+      .flatMap(storeNames => updateStores(storeNames))
+      .map(results => s"Updated all bags of ${ results.size } stores ($vaultBaseUri)")
+  }
+
+  def initSingleStore(storeName: String): Try[FeedBackMessage] = {
+    getBagIds(storeName)
+      .flatMap(bagIds => updateBags(storeName, bagIds))
```

Comment:
either

`.flatMap(updateBags(storeName, _))`

or curry `updateBags` and do

`.flatMap(updateBags(storeName))`